### PR TITLE
Fix profile name not appearing in Vue profile streams

### PR DESCRIPTION
**Pull request by christophehenry**
Failing Django tests. @jaywink Need halp!

halp_2

### CODE REVIEW

File: socialhome/users/views.py
Comment (jaywink) [inline socialhome/users/views.py]: I guess you're right about using the profile serializer to serialize the data for the frontend, but the trouble with adding all the extra fields there is that it will make the profile API a lot slower. But, since we disabled profile list API view totally some time ago, I guess performance isn't really an issue.

However, using `LimitedProfileSerializer` for unauthenticated users is wrong. It's not meant to control visibility, but to contain less data, ie the data required to render a grid item. A serializer doesn't itself do anything about visibility - that is what the view is for. If using a serializer directly, one must ensure only those fields are coming out that the user can see.

Our `ProfileSerializer` contains only data that is available always if the user can see the profile, so this is ok. Just remove the `else` and the usage of the limited serializer.

File: socialhome/users/serializers.py
Comment (jaywink) [inline socialhome/users/serializers.py]: No need for `safer_*` image url method usage any more - those will soon be removed, since #348. And the API already has `image_url_large`.

File: socialhome/users/serializers.py
Comment (jaywink) [inline socialhome/users/serializers.py]: No need for `str` - a count should be a number.

File: socialhome/users/serializers.py
Comment (jaywink) [inline socialhome/users/serializers.py]: No need for `str` - a count should be a number.

File: socialhome/users/serializers.py
Comment (jaywink) [inline socialhome/users/serializers.py]: `obj.user` should be the user *viewing the page*, not the user linked to profile. The `Content` queryset uses this to check what content is visible. For example, the profile might have pinned content which is not visible to the viewing user.

See [this serializer method in Content serializer](https://github.com/jaywink/socialhome/blob/bf8dc7af5a4cfe4f4d8e73aa6aa2464446818961/socialhome/content/serializers.py#L83) on how to get the user from the request object.

Comment (jaywink) [reply socialhome/users/serializers.py]: No need for `pinned=True`, the `profile_pinned()` queryset method already does that.

File: socialhome/users/serializers.py
Comment (jaywink) [inline socialhome/users/serializers.py]: I would prefer maybe `has_pinned_content` but either way is fine.

File: socialhome/users/serializers.py
Comment (jaywink) [inline socialhome/users/serializers.py]: No need for this as commented that we shouldn't use this serializer at all.

**Review by jaywink — CHANGES_REQUESTED** [review]
Some things to tweak. Thanks for working on these!

Commit message should be a bit more clear on what is done, ie the refactoring of view json -> serializer would be nice to be mentioned. Also you're adding several properties to the Profile API which should be mentioned. Also, commits should reference an issue like this:

`Refs: #202`. For good commit messages, see https://chris.beams.io/posts/git-commit/

**Comment by jaywink** [comment]
Oh I think the test wont crash any more if you pass in the correct user to `profile_pinned()` 👍

**Comment by christophehenry** [comment]
> For good commit messages, see https://chris.beams.io/posts/git-commit/

I'm not good at writing informative git commit messages. 😢

**Comment by jaywink** [comment]
> I'm not good at writing informative git commit messages. 😢

.. but learning new things or to be better at something is why we have these reviews!

File: socialhome/users/views.py
Comment (christophehenry) [reply socialhome/users/views.py]: > But, since we disabled profile list API view totally some time ago, I guess performance isn't really an issue.

I prefer using one unique source rather than multiple. If there's a serializer for the API, better using it. What the front-end needs, any other application may need it. So, is it a go or a no-go?

Comment (jaywink) [reply socialhome/users/views.py]: Sure, but we must also think of performance. As said, this is fine, since we only fetch individual objects, not lists. If we do in the future, we'll need to precalculate the values on the model. For now, this is fine.

File: socialhome/users/tests/test_views.py
Comment (jaywink) [inline socialhome/users/tests/test_views.py]: This change is unnecessary - `self.assertEqual` calls `self.assertDictEqual` when comparing dict.

**Review by jaywink — CHANGES_REQUESTED** [review]

File: socialhome/users/serializers.py
Comment (jaywink) [inline socialhome/users/serializers.py]: Please place alphabetically.

File: socialhome/users/serializers.py
Comment (jaywink) [inline socialhome/users/serializers.py]: Please place alphabetically.

File: socialhome/users/serializers.py
Comment (jaywink) [inline socialhome/users/serializers.py]: `.filter()` is unnecessary here.

File: socialhome/users/tests/test_views.py
Comment (jaywink) [inline socialhome/users/tests/test_views.py]: I don't think `view.pinned_content_exists` should exist any more since we use the serializer now?

File: socialhome/users/views.py
Comment (jaywink) [inline socialhome/users/views.py]: This should be removed AFAICT - we always want to serialize the profile.

File: socialhome/users/tests/test_views.py
Comment (christophehenry) [reply socialhome/users/tests/test_views.py]: It's used on the Vue side.

Comment (jaywink) [reply socialhome/users/tests/test_views.py]: >  <b-button v-if="profile.has_pinned_content" 

I think you renamed it already in this PR :) Also, I was referring to the Django View. So [this](https://github.com/jaywink/socialhome/blob/6559f3ece9c7658ea797d2e01726e5c0ec24770d/socialhome/users/views.py#L106) and [this](https://github.com/jaywink/socialhome/blob/6559f3ece9c7658ea797d2e01726e5c0ec24770d/socialhome/users/views.py#L36) can be removed.

File: socialhome/users/views.py
Comment (christophehenry) [reply socialhome/users/views.py]: Yes, that's right.

File: socialhome/users/tests/test_views.py
Comment (christophehenry) [reply socialhome/users/tests/test_views.py]: Won't it break the jQuery code?

**Comment by christophehenry** [comment]
Fixing remarks and tests done. Waiting for another review.

File: socialhome/users/tests/test_views.py
Comment (jaywink) [inline socialhome/users/tests/test_views.py]: Test is failing. Maybe because you're using a user ID, not profile ID?

File: socialhome/users/tests/test_views.py
Comment (jaywink) [inline socialhome/users/tests/test_views.py]: Same here?

Comment (jaywink) [reply socialhome/users/tests/test_views.py]: You can also probably remove the `view.followers_count = 3` from this test slightly above line 385.

File: socialhome/users/tests/test_views.py
Comment (jaywink) [inline socialhome/users/tests/test_views.py]: Same here

Comment (jaywink) [reply socialhome/users/tests/test_views.py]: You can also probably remove the view.followers_count = 3 from this test slightly above line 389.

File: socialhome/users/views.py
Comment (jaywink) [inline socialhome/users/views.py]: You can remove `LimitedProfileSerializer` import.

**Review by jaywink — CHANGES_REQUESTED** [review]
A few more tweaks + fixing of failing tests

File: socialhome/users/tests/test_views.py
Comment (christophehenry) [reply socialhome/users/tests/test_views.py]: Shouldn't be the same?

Comment (jaywink) [reply socialhome/users/tests/test_views.py]: No, local users have a `User`. Remote users don't. But everybody has a `Profile`. So what you should be setting (assuming a user is logged in) is `request.user.profile_id`. If it's an anonymous user, then pass in `None`, since anonymous users don't have a profile.

**Comment by christophehenry** [comment]
Done. Come again, I'm waiting you ! >:D

File: socialhome/streams/app/components/stamped_elements/ProfileStampedElement.vue
Comment (jaywink) [inline socialhome/streams/app/components/stamped_elements/ProfileStampedElement.vue]: Why switch to medium? Large is 300px which is what is currently in the profile streams. Medium is 100px which is too small imho.

File: socialhome/users/tests/test_views.py
Comment (jaywink) [reply socialhome/users/tests/test_views.py]: This doesn't seem right. If the test passes it's probably because the ID's happen to match, not because they will always match. `currentBrowsingProfileId` should reference a `Profile` object ID. Does the test pass if you set it as `request.user.profile_id`?

File: socialhome/users/tests/test_views.py
Comment (jaywink) [reply socialhome/users/tests/test_views.py]: Same here, should not be `user.id`. Also, what is weird that we're using `AnonymousUser`, so this test should probably fail :/ I'll have a quick look at why it's not.

**Review by jaywink — CHANGES_REQUESTED** [review]
A few minor comments.

File: socialhome/users/tests/test_views.py
Comment (jaywink) [reply socialhome/users/tests/test_views.py]: Ah, this is why:

```
In [4]: AnonymousUser().id is None
Out[4]: True
```

Still, this would be better more explicit to say `"currentBrowsingProfileId": None,` as it was before.

File: socialhome/users/tests/test_views.py
Comment (jaywink) [reply socialhome/users/tests/test_views.py]: Ah, this is why:

```
In [4]: AnonymousUser().id is None
Out[4]: True
```

Still, this would be better more explicit to say `"currentBrowsingProfileId": None,` as it was before.

File: socialhome/users/tests/test_views.py
Comment (christophehenry) [reply socialhome/users/tests/test_views.py]: `AnonymousUser` has no attribute `profile_id`. If I put `None` here, the test passes. I suppose `request.user.id` is `None` when using `AnonymouseUser`.

Comment (jaywink) [reply socialhome/users/tests/test_views.py]: Yeah the test should just check for the values it expects so just compare to `None` 👍

**Review by jaywink — APPROVED** [review]

**Comment by jaywink** [comment]
Thanks! <3

**Pull request merged**
